server/controllers/finance/patientInvoice.js

```diff
@@ -290,3 +299,45 @@ function reference(req, res, next) {
   .catch(next)
   .done();
 }
+
+
+/**
+* GET /invoices/patient/report
+* Returns a pdf file for Patient Invoice
+*
+* @function getPatientInvoice
+*/
+function getPatientInvoice(req, res, next) {
+  const request = {
+    query : req.query,
+    enterprise : req.session.enterprise,
+    project : req.session.project
+  };
+
+  let invoiceListQuery =
```

Comment:
This code is the exact same code as [the patientInvoice list() function](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/server/controllers/finance/patientInvoice.js#L53).  Since it is a duplication, I propose that you use the same function.

I propose that you make a new function in the `patientInvoice` file called `listInvoices()`.  You could then write:
```js
// GET /invoices
function list(req, res, next) {
  listInvoices()
  .then(function (invoices) {
    res.status(200).json(invoices);
  })
  .catch(next)
  .done();
}

// GET /invoices/patient/report
function getPatientInvoice(req, res, next) {

  const request = { /* some values */ };

  listInvoices()
  .then(invoices => listReceipt.build(invoices, request):
   // ... etc ...
  })
  .catch(next)
  .done();
}
```

Remember, in software engineering, we [don't like to repeat ourselves](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).